app/src/main/java/ch/blinkenlights/android/vanilla/CoverCache.java

```diff
@@ -463,6 +468,19 @@ public Bitmap createBitmap(Song song, long maxPxCount) {
 					}
 				}
 
+				if (inputStream == null && (CoverCache.mCoverLoadMode & CoverCache.COVER_MODE_INLINE) != 0)
+				{
+					android.media.MediaMetadataRetriever mmr = new MediaMetadataRetriever();
+					mmr.setDataSource(song.path);
+
+					byte[] data = mmr.getEmbeddedPicture();
```

Comment:
How fast is this? I know that the tag reader can be pretty slow.
Does it take 1ms or more like 100ms to extract a cover like this?

Reply:
In Android Studio (AVD) we are talking 10-20ms for my test albums.